### PR TITLE
Individual animations for insert/remove/update

### PR DESCRIPTION
**Pull request by bstien**
# What
You should be able to select individual animations for the different updates to the tableview.

### CODE REVIEW

File: Diffuse/Extensions/UITableViewExtensions.swift [hunk 1]
```diff
@@ -6,7 +6,9 @@ import UIKit
 
 extension UITableView {
     public func reload(with changes: CollectionChanges,
-                       animation: UITableView.RowAnimation = .automatic,
+                       insertAnimation: UITableView.RowAnimation = .automatic,
+                       reloadAnimation: UITableView.RowAnimation = .automatic,
+                       deleteAnimation: UITableView.RowAnimation = .automatic,
                        section: Int = 0,
```
Comment (shredlocker) [inline Diffuse/Extensions/UITableViewExtensions.swift:12]: Maybe use a dictionary with enum keys for this?

**Review by robinsalehjan — APPROVED** [review]

File: Diffuse/Extensions/UITableViewExtensions.swift (hunk 1 above)
Comment (bstien) [reply Diffuse/Extensions/UITableViewExtensions.swift:12]: You mean something like this?

```swift
enum TableViewOperation {
	case insert
	case update
	case remove
}

func reload( animations: [TableViewOperation: UITableView.RowAnimation] ) {}
```

Comment (shredlocker) [reply Diffuse/Extensions/UITableViewExtensions.swift:12]: Yep, something like that :)

Comment (bstien) [reply Diffuse/Extensions/UITableViewExtensions.swift:12]: I'll give it a try :+1:

Comment (bstien) [reply Diffuse/Extensions/UITableViewExtensions.swift:12]: But perhaps in another PR so we can discuss it. I'll merge this and open another one with these changes

Comment (shredlocker) [reply Diffuse/Extensions/UITableViewExtensions.swift:12]: 👍

Comment (bstien) [reply Diffuse/Extensions/UITableViewExtensions.swift:12]: The changes are in PR #6

**Pull request merged**